test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -569,6 +583,27 @@ private void afterInternal(boolean afterClass) throws Exception {
         }
     }
 
+    public void unblockRepositories() throws Exception {
```

Comment:
I'm a little unsure about this. I think we should only run this if the original test failed shouldn't we? (this is what we do in REST tests)
Otherwise we can have tests leaking running/blocked snapshots in the background and are quietly cleaning them up here?

Reply:
Yes, that makes sense. do we have something like `getSuiteFailureMarker()` but for a single test? I cannot find any method like that on `ESTestCase`

Reply:
Maybe `org.elasticsearch.test.ESTestCase#afterIfFailed` will work here (I haven't checked the exact order of things I must admit). If it's called too late maybe we can do something with a `@Rule`? (I must admit I'm not an expert in JUnit so the latter is a bit of a guess).